big_tests/tests/rdbms_SUITE.erl

```diff
@@ -387,6 +394,48 @@ test_request_insert(Config) ->
                            selected_to_sorted(SelectResult))
       end, ok, #{name => request_queries}).
 
+test_wrapped_request(Config) ->
+    % given
+    erase_table(Config),
+    sql_prepare(Config, insert_one, test_types, [unicode],
+                <<"INSERT INTO test_types(unicode) VALUES (?)">>),
+    rpc(mim(), mongoose_metrics, ensure_metric, [global, [test_metric], histogram]),
+    WrapperFun = fun(SqlExecute) ->
+        {Time, Result} = timer:tc(SqlExecute),
+        mongoose_metrics:update(global, [test_metric], Time),
+        Result
+    end,
+
+    % when
+    sql_execute_wrapped_request(Config, insert_one, [<<"check1">>], WrapperFun),
+
+    % then
+    mongoose_helper:wait_until(
+        fun() ->
+            SelectResult = sql_query(Config, "SELECT unicode FROM test_types"),
+            ?assertEqual({selected, [{<<"check1">>}]}, selected_to_sorted(SelectResult))
+        end, ok, #{name => request_queries}),
+
+    {ok, Metric} = rpc(mim(), mongoose_metrics, get_metric_value, [global, [test_metric]]),
+    MetricValue = proplists:get_value(mean, Metric),
+    ?assert(MetricValue > 0).
+
+test_failed_wrapper(Config) ->
+    % given
+    erase_table(Config),
+    sql_prepare(Config, insert_one, test_types, [unicode],
+                <<"INSERT INTO test_types(unicode) VALUES (?)">>),
+    ok = rpc(mim(), meck, new, [supervisor, [passthrough, no_link, unstick]]),
+    WrapperFun = fun(_SqlExecute) ->
+        error(wrapper_crashed)
+    end,
+
+    % when
+    sql_execute_wrapped_request(Config, insert_one, [<<"check1">>], WrapperFun),
+
```

Comment:
U also need to check what `gen_server:wait_response` returns I think in this test. (because a request is an async operation, so just checking history is not reliable, the crash could not be in the history just yet).

Ideally we would wanna check that the same worker can still process other operations, but we don't support sending queries to a specific worker, sadly.

Reply:
Done. I had to do a little hack because `gen_server:wait_response` has to be called by the same process and according to documentation I can't make any assumption about the process which is handling rpc call (i.e. calling `apply`).